doc/script_commands.txt

```diff
@@ -5317,6 +5317,27 @@ Check getitem2() to understand its expanded parameters.
 
 ---------------------------------------
 
+*delitemidx(<index>{, <amount>{, <account id>}})
+
+This command will remove the item at the given inventory index.
+
+If amount is not specified, it will remove all of the item.
+
+See getinventorylist() for an explanation on retrieving an inventory index.
+
+It is a good idea to check if the player actually has an item at the
+specified index. After deleting an item at an index, that index can
+remain empty until the player re-logs. If you try to delete an item at
+an invalid index or an index with no item, the script will be terminated
+with an error.
+
+Examples:
+
+//This will attempt to delete the item at inventory index 45.
+	delitemidx(45);
```

Comment:
I think this is bad example

```
//	This will remove all invalid Item ID in player's inventory
	getinventorylist();
	for (.@i = 0; .@i < @inventorylist_count; ++.@i)
		if (getiteminfo(@inventorylist_id[.@i], ITEMINFO_TYPE) == -1)
			delitemidx(.@i);
```


Note: I have already tested with
```
prontera,152,188,5	script	test getitemname2	1_F_MARIA,{
	getitem2 1501, 1,1,0,0, 0,0,0,0;
	getitem2 1501, 1,1,1,0, 0,0,0,0;
	getitem2 1501, 1,1,2,0, 0,0,0,0;
	getitem 1201, 1;
	getitem2 1502, 1,1,0,0, 4001,0,0,0;
	getitem2 1501, 1,1,0,0, 4001,0,0,0;
	getinventorylist();
	for (.@i = 0; .@i < @inventorylist_count; ++.@i) {
		if (getiteminfo(@inventorylist_id[.@i], ITEMINFO_TYPE) == IT_WEAPON)
			delitemidx(.@i);
	}
	end;
}
```